.github/actions/discover_python_version/action.yml

```
@@ -0,0 +1,25 @@
+name: Discover minimum Python version
+description: Discover the lowest supported Python version based on pyptoject.toml file
+
+outputs:
+  pyversion:
+    description: Lowest supported python version
+    value: ${{ steps.pyversion.outputs.pyver }}
+
+runs:
+  using: composite
+  steps:
+    - name: extract Python version
+      id: pyversion
+      shell: bash
+      run: |
+        # try to extract the 'python = "xxx"' string from TOML file
+        # this script will fail in following situations:
+        #  - the file is not there
+        #  - the string is not there, so no pattern for Python is provided
+        #  - version is not readable - this is handled by '/python/q1' SED command,
+        #    when the regex does not match SED won't do anything and it will return the
+        #    full string, this command looks for 'python' and return exit code 1 if it finds one
+
+        PYVER=$(grep '^[Pp]ython ' pyproject.toml | sed -E 's/python = \"\^?([0-9]\.[0-9])(\.[0-9])?\"/\1/g; /python/q1')
```

I think this doesn't match 3.10.1 or 3.8.10 when the digit is 10 or higher. Maybe we can update like below?

```suggestion
        PYVER=$(grep '^[Pp]ython ' pyproject.toml | sed -E 's/python = \"\^?(\d\.\d+)(\.\d+)?\"/\1/g; /python/q1')
```